markdown/renderer.go

```diff
@@ -23,26 +23,54 @@ var (
 	codeBlockChars          = []byte("```")
 	tableHeaderColChar      = []byte{'-'}
 	tableHeaderAlignColChar = []byte{':'}
+	heading1UnderlineChar   = []byte{'='}
+	heading2UnderlineChar   = []byte{'-'}
 )
 
 // Ensure compatibility with Goldmark parser.
 var _ renderer.Renderer = &Renderer{}
 
 // Renderer allows to render markdown AST into markdown bytes in consistent format.
 // Render is reusable across Renders, it holds configuration only.
-type Renderer struct{}
+type Renderer struct {
+	Opts *renderer.Config
+}
 
 func (mr *Renderer) AddOptions(opts ...renderer.Option) {
-	c := renderer.Config{}
+	// this function is a bit of a bad design, as the options here are goldmark's HTML renderer
+	// structure... but goldmark just forwards it to us anyway
+	// (but it's goldmark's bad design, not ours :))
 	for _, o := range opts {
-		o.SetConfig(&c)
+		o.SetConfig(mr.Opts)
 	}
+}
 
-	// TODO(bwplotka): Add headers optionality (https://github.com/Kunde21/markdownfmt/issues/14).
+type UnderlineHeadings struct {
```

Comment:
Can we have consistency with other options in goldmark? Something like 

```
func WithUnderlineHeadings() witthUnderlineHeadings { return witthUnderlineHeadings{} }

type witthUnderlineHeadings struct {}

func (u witthUnderlineHeadings) SetConfig(config *renderer.Config) {  ... etc
```

Reply:
oh yeah I should have done that, brain fart